drivers/wifi/esp8266/esp8266.c

```diff
@@ -0,0 +1,844 @@
+/*
+ * Copyright (c) 2019 Tobias Svehagen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_LEVEL CONFIG_WIFI_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wifi_esp8266);
+
+#include <kernel.h>
+#include <ctype.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <device.h>
+#include <init.h>
+#include <stdlib.h>
+
+#include <drivers/gpio.h>
+
+#include <net/net_if.h>
+#include <net/net_offload.h>
+#include <net/wifi_mgmt.h>
+
+#include "esp8266.h"
+
+/* pin settings */
+enum modem_control_pins {
+#if defined(DT_INST_0_ESPRESSIF_ESP8266_WIFI_RESET_GPIOS_PIN)
+	WIFI_RESET,
+#endif
+	NUM_PINS,
+};
+
+static struct modem_pin modem_pins[] = {
+#if defined(DT_INST_0_ESPRESSIF_ESP8266_WIFI_RESET_GPIOS_PIN)
+	MODEM_PIN(DT_INST_0_ESPRESSIF_ESP8266_WIFI_RESET_GPIOS_CONTROLLER,
+		  DT_INST_0_ESPRESSIF_ESP8266_WIFI_RESET_GPIOS_PIN,
+		  GPIO_DIR_OUT),
```

Comment:
esp8266.c:39:13: warning: Macro is deprecated
   39 |     GPIO_DIR_OUT),
      |             ^~~~~~